docs/cruisecontroloperation.md

```diff
@@ -0,0 +1,320 @@
+
+```yaml
+kind: CruiseControlOperation
+metadata:
+...
+  name: kafka-addbroker-mhh72
+  labels:
+    pause: "true"
+...
+```
+
+- You can set automatic cleanup time for the created `CruiseControlOperations` in the `KafkaCluster` custom resource. In the above example the finished (completed successfully or completedWithError and errorPolicy: ignore) `CruiseControlOperation` custom resources will be deleted after the 300 sec elapsed.
+
+```yaml
+apiVersion: kafka.banzaicloud.io/v1beta1
+kind: KafkaCluster
+...
+spec:
+...
+  cruiseControlConfig:
+    cruiseControlOperationSpec:
+      ttlSecondsAfterFinished: 300
+...
+```
+
+### Example for the ignore and pause use-cases
+
+1. In the first example we extended the Kafka cluster with one broker. Now we will remove 2 brokers at the same time by editing the `KafkaCluster` custom resource.
+
+```yaml
+ Spec:
+ ...  
+   brokers:
+    - id: 0
+      brokerConfigGroup: "default"
+    - id: 1
+      brokerConfigGroup: "default"
+```
+
+2. There will be `remove_broker` operations for both brokers (kafka-removebroker-lg7qm, kafka-removebroker-4plfq). The first one is already in running state.
+
+```yaml
+status:
+...
+  brokersState:
+        "2": {
+        ...
+            "gracefulActionState": {
+                "cruiseControlOperationReference": {
+                    "name": "kafka-removebroker-lg7qm"
+                },
+                "cruiseControlState": "GracefulDownscaleRunning"
+            },
+        ...
+        },
+        "3": {
+        ...
+            "gracefulActionState": {
+                "cruiseControlOperationReference": {
+                    "name": "kafka-removebroker-4plfq"
+                },
+                "cruiseControlState": "GracefulDownscaleScheduled"
+            },
+        ...
+        }
+```
+
+3. We assume that something unexpected happened this time so the remove_broker operation will get `GracefulDownscaleCompletedWithError` state.
+
+```yaml
+status:
+...
+  brokersState:
+        "2": {
+        ...
+            "gracefulActionState": {
+                "cruiseControlOperationReference": {
+                    "name": "kafka-removebroker-lg7qm"
+                },
+                "cruiseControlState": "GracefulDownscaleCompletedWithError"
+            },
+        ...
+        },
+        "3": {
+        ...
+            "gracefulActionState": {
+                "cruiseControlOperationReference": {
+                    "name": "kafka-removebroker-4plfq"
+                },
+                "cruiseControlState": "GracefulDownscaleScheduled"
+            },
+        ...
+        }
+```
```

Comment:
**Question**: Why are we using JSON code inside YAML here? (JSON is less eye-friendly, but syntactically valid, just not sure whether it is intentional.)

Same for the other blocks below containing braces and double quotes.

Reply:
It was a mistake. Should I rewrite those into yaml?